apps/typescriptcourse/src/templates/portable-text-article-template.tsx

```diff
@@ -58,14 +67,37 @@ const ArticleTemplate: React.FC<ArticleTemplateProps> = ({
           </div>
         </div>
         <section data-article="">
-          {getCTA({subscribersOnly, hasSubscribed, cta, router})}
+          {cta ? (
+            <CTAContainer>
+              <div className="pb-8 flex flex-col items-center prose prose-p:max-w-[30ch] prose-headings:font-bold text-center">
+                <PortableText
+                  value={cta.body}
+                  components={PortableTextComponents}
+                />
+              </div>
+              <SubscribeToConvertkitForm
+                className="article"
+                formId={cta.formId}
+                onSuccess={(subscriber: any) => {
+                  if (subscriber) {
+                    const redirectUrl = redirectUrlBuilder(
+                      subscriber,
+                      '/confirm',
+                    )
+                    router.push(redirectUrl)
+                  }
+                }}
+                actionLabel={cta.actionLabel}
+              />
+            </CTAContainer>
```

Comment:
below is a solid apparatus for switching CTAs, but I want to put a little more thought into it so I simplified down to just the one CTA that isn't smart around the user being a subscriber or not

one thing that's interesting is that this section doesn't need to be server rendered and we can do client side decision making loading it dynamically

if we want to hide half the article, we should consider middleware and rewrites IMO